### PR TITLE
Tileset reader

### PR DESCRIPTION
**Pull request by LorenzoMarnat**
This PR introduce the Tileset Reader, which allows to read, transform and/or merge 3DTiles tilesets. It comes with its doc and tests.

This PR also reworks the Node/Tree relation used to create the tileset. The LodNode/LodTree now inherits from GeometryNode/GeometryTree. This allows to create a tree with nodes and geometries differently. In the TilesetReader, the TileHierarchy (inheriting from GeometryTree) creates a tree based on the tiles parent/child hierarchy.

Note: this PR partially answers [this Py3dTiles issue](https://github.com/VCityTeam/py3dtiles/issues/15).

### CODE REVIEW

File: py3dtilers/Common/tileset_creation.py [hunk 1]
```diff
@@ -1,19 +1,17 @@
 import numpy as np
 from py3dtiles import B3dm, BatchTable, BoundingVolumeBox, GlTF, GlTFMaterial
```
Comment (EricBoix) [inline py3dtilers/Common/tileset_creation.py:2]: The three functions are bound together and only create_tileset() is called from other portions of the code.
Those functions could be static methods of a (static) class that is indeed a function aggregator whose purpose is to convert a GeometryTree instance to a TileSet instance. Following this logic, we could 
* call such a static class e.g. `FromGeometryTreeToTileset`
* rename create_tileset(...) to convert(...)
* make create_tile() and create_tile_content() private (with the conventional `__` prefix)

and then use that classe by writing something like
`tileset = FromGeometryTreeToTileset.convert(my_objects_to_tile_instance, that_extension)`

File: py3dtilers/TilesetReader/parsed_b3dm.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import os
+
```
Comment (EricBoix) [inline py3dtilers/TilesetReader/parsed_b3dm.py:2]: Same comment as for tileset_creation.py (allthough these are already classes).
The only exposed method should here be ParsedB3dm::parse_tile() that could be renamed to convert().

File: py3dtilers/TilesetReader/parsed_b3dm.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+from py3dtiles import TriangleSoup, GlTFMaterial
+from ..Common import ObjectsToTile, ObjectToTile
+from ..Texture import Texture
+
+
+class ParsedB3dm(ObjectToTile):
```
Comment (EricBoix) [inline py3dtilers/TilesetReader/parsed_b3dm.py:8]: Again (and sorry for that) the semantics carried by this class name is not very helpful.
Maybe `FromTileToObjetToTile` would be clearer ? (although it is still awkward: anything better ?)

File: py3dtilers/TilesetReader/parsed_b3dm.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    def parse_triangle_soup(self, triangle_soup, materials, tile_index=0):
+        """
+        Parse the triangle soup to re-create the geometries.
+        :param triangle_soup: the triangle soup
+        :param materials: the materials of the tile
+        :param tile_index: the index of the tile
+
+        :return: a list of geometries
+        """
+        triangles = triangle_soup.triangles[0]
```
Comment (EricBoix) [inline py3dtilers/TilesetReader/parsed_b3dm.py:83]: Man, the `TriangleSoup` class interface is really in need of some love.
Maybe we could suggest to Py3dTiles to e.g rename `.triangles[1]` accessor to a `get_vertex_ids()` method ?

File: py3dtilers/TilesetReader/TilesetReader.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
+import sys
+
+from py3dtiles import TilesetReader
+from .parsed_b3dm import ParsedB3dms
+from .tile_hierarchy import TileHierarchy
+from ..Common import Tiler, create_tileset
+
+
+class B3dmTiler(Tiler):
```
Comment (EricBoix) [inline py3dtilers/TilesetReader/TilesetReader.py:9]: Semantics again: maybe naming that class `3DTilesImporter` would clarify the fact that this class is inherently a treatment ? (just as converters are: refer above)

File: py3dtilers/TilesetReader/tile_hierarchy.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
+from ..Common import GeometryNode, GeometryTree, ObjectsToTile
+
+
+class TileHierarchy(GeometryTree):
```
Comment (EricBoix) [inline py3dtilers/TilesetReader/tile_hierarchy.py:4]: Still semantics: if we follow the Tree/Node terminology this is a TileSetTree. No ?

**Review by EricBoix — CHANGES_REQUESTED** [review]

File: py3dtilers/TilesetReader/parsed_b3dm.py (hunk 4 above)
Comment (LorenzoMarnat) [reply py3dtilers/TilesetReader/parsed_b3dm.py]: I completely agree. `.triangles` is a list containing the triangles, UVs, ids, and whatever the user puts in it. We should more getters/attributes in the `TriangleSoup` class in py3dtiles.

**Review by EricBoix — APPROVED** [review]
Opened some unanswered issues of this PR (mainly concerning structural clarification) that must be answered prior to other issues.

**Pull request merged**